service/header/sync.go

```diff
@@ -0,0 +1,120 @@
+package header
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+// syncer implements simplest possible synchronization for headers.
+type syncer struct {
+	exchange Exchange
+	store    Store
+
+	// done is triggered when syncing is finished
+	// it assumes that syncer is only used once
+	done chan struct{}
+}
+
+func newSyncer(exchange Exchange, store Store) *syncer {
+	return &syncer{
+		exchange: exchange,
+		store:    store,
+		done:     make(chan struct{}),
+	}
+}
+
+// Sync syncs all headers up to the latest known header in the network.
+func (s *syncer) Sync(ctx context.Context) {
+	log.Info("syncing headers")
+	// TODO(@Wondertan): Retry logic
+	for {
+		localHead, err := s.store.Head(ctx)
+		if err != nil {
+			log.Errorw("getting local head", "err", err)
+			return
+		}
+
+		netHead, err := s.exchange.RequestHead(ctx)
+		if err != nil {
+			log.Errorw("requesting network head", "err", err)
+			return
+		}
+
+		if localHead.Height >= netHead.Height {
```

Comment:
Can we go over the netHead's height? e.g. is there really a case where `>` would be true?

Reply:
No, not possible